src/org/intellij/erlang/ErlangReferenceContributor.java

```diff
@@ -29,30 +32,40 @@
 
   @Override
   public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar) {
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  2, 1);
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn",      3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn",      3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn_link", 3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn_link", 3, 1);
+    registerRecords(registrar, "erlang", "is_record", 2, 1);
+    registerRecords(registrar, "erlang", "is_record", 3, 1);
+
+    registrar.registerReferenceProvider(psiElement(ErlangQAtom.class).with(
+      new ErlangFunctionCallSpecParameter<PsiElement>()), getProvider(Kind.MODULE));
+
+    ErlangFunctionCallNamedParameter<PsiElement> funParameter =
+      new ErlangFunctionCallNamedParameter<PsiElement>("Function", "Fun", "_Function");
+    registrar.registerReferenceProvider(psiElement(ErlangQAtom.class).with(funParameter), getProvider(Kind.FUNCTION));
```

Comment:
Please, clarify your logic - you have to check you're looking at a parameter expression of interest - that is ErlangMaxExpression, and only then check that it's a properly named argument.